src/CSV/CSVWriter.php

```diff
@@ -35,6 +36,10 @@ class CSVWriter
         'write_utf8_bom' => FALSE
     ];
 
+    public function __construct(private readonly bool $quote_headers = true)
```

Comment:
I think this should be part of the options to `open` for consistency with the `write_utf8_bom` option - as you'd want to specify when writing the file rather than requiring a separate DI service with a different configuration?